src/dune.ml

```diff
@@ -0,0 +1,38 @@
+open Import
+
+type t =
+  { root : Path.t
+  ; bin : Cmd.spawn
+  }
+
+let binary = Path.of_string "dune"
+
+let make ~root () =
+  let open Promise.Syntax in
+  (* Should we do something specific for Windows ? *)
+  let spawn = { Cmd.bin = binary; args = [] } in
+  let+ spawn = Cmd.check_spawn spawn in
+  match spawn with
+  | Ok bin -> Some { bin; root }
+  | Error _ -> None
+;;
+
+let detect_dune_lock_dir t =
+  (* Path to the dune.lock dir *)
+  let dune_lock_path = Path.join t.root (Path.of_string "dune.lock") in
+  Fs.exists (Path.to_string dune_lock_path)
+;;
+
+let detect_dune_ocamllsp t =
+  (* Path to the ocaml-lsp-server.pkg file *)
+  let ocamllsp =
+    Path.join
+      t.root
+      (Path.join (Path.of_string "dev-tools.locks") (Path.of_string "ocaml-lsp-server"))
+  in
+  Fs.exists (Path.to_string ocamllsp)
+;;
+
+let exec t ~args = Cmd.Spawn (Cmd.append t.bin args)
```

Comment:
This function is very generic. It's meant to execute any dune command. In my opinion, it'd be better to have different functions: One to execute a tool via dune (`<dune bin> exec tools <tool>`) and one to lock a project. The reasons:
- If the current preview dune API changes, we only need to adapt to that here in dune.ml and not all over the code base.
- That'd match the abstraction of the other sandboxes.